project/products/routes.py

```diff
@@ -0,0 +1,176 @@
+        flash('New Product {} Successfully Created'.format(newProduct.name))
+        return redirect(url_for('categories.showCategoryProducts',
+                                cat_id=newProduct.category_id))
+    else:
+        return render_template('addproduct.html',
+                               categories=(category_service
+                                           .get_all_categories()),
+                               isLogin=auth_service.is_user_authorized())
+
+
+@products.route('/editproduct/<int:cat_id>/<int:prod_id>/',
+                methods=['GET', 'POST'])
+def editProduct(cat_id, prod_id):
+    """App route function to edit existing Product."""
+    # If user is not logged in, inform him about it and redirect
+    if 'email' not in login_session:
+        flash("You need to Log In if you want to edit")
+        return redirect('/login')
+    # Get Product that user wants to edit
+    productToEdit = session.query(Product).filter_by(id=prod_id).one()
+    # Check if user is Creator, if not inform that he cannot
+    # do changes
+    if not login_session['email'] == productToEdit.user.email:
+        flash("You need to be a Creator of the Product to edit it")
+        return redirect(url_for('products.showProduct',
+                                cat_id=cat_id, prod_id=prod_id))
+    if request.method == 'POST':
+        if request.form['name']:
+            productToEdit.name = request.form['name']
+        if request.form['image_url']:
+            productToEdit.image_url = request.form['image_url']
+        if request.form['product_url']:
+            productToEdit.product_url = request.form['product_url']
+        if request.form['description']:
+            productToEdit.description = request.form['description']
+        session.add(productToEdit)
+        session.commit()
+        flash('You successfully edited product')
+        return redirect(url_for('products.showProduct',
+                                cat_id=cat_id, prod_id=prod_id))
+    else:
+        return render_template('editproduct.html',
+                               categories=(category_service
+                                           .get_all_categories()),
+                               isLogin=auth_service.is_user_authorized(),
+                               category=(category_service
+                                         .get_category_by_id(cat_id)),
+                               productToEdit=productToEdit)
+
+
+@products.route('/deleteproduct/<int:cat_id>/<int:prod_id>/',
+                methods=['GET', 'POST'])
+def deleteProduct(cat_id, prod_id):
+    """App route function to delete existing product."""
+    # If user is not logged in, inform him about it and redirect
+    if 'email' not in login_session:
+        flash("You need to Log In if you want to delete product")
+        return redirect('/login')
+    # Get the Product that User wants to delete
+    productToDelete = session.query(Product).filter_by(id=prod_id).one()
```

Comment:
You did a good job moving most of DB calls to services! However, there should be no two ways to get objects of a certain type. You either always use a service (preferred) or use queries against models directly (better to avoid). For ex (you already have this method in your service):
```suggestion
    productToDelete = product_service.get_product_by_id(prod_id)
```
Also, it's a good habit to spell out words in full in variable names. I.e. `product_id` instead of `prod_id`, `category_name` instead of `cat_name`, etc.